infra/chromadb_compaction/tests/integration/test_end_to_end_workflow.py

```
@@ -0,0 +1,475 @@
+        # Test S3 failure scenario
+        # Delete the bucket to simulate S3 failure
+        s3.delete_bucket(Bucket="test-chromadb-bucket")
+        
+        # Try to create a snapshot (should fail)
+        try:
+            s3.put_object(
+                Bucket="test-chromadb-bucket",
+                Key="test/snapshot/test.txt",
+                Body="test content"
+            )
+            assert False, "Should have failed due to missing bucket"
```

Ruff (0.14.1)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)